docs/plat/allwinner.rst

```diff
@@ -0,0 +1,26 @@
+Trusted Firmware-A for Allwinner ARMv8 SoCs
+===========================================
+
+Trusted Firmware-A (TF-A) implements the EL3 firmware layer for Allwinner
+SoCs with ARMv8 cores. Only BL31 is used to provide proper EL3 setup and
+PSCI runtime services.
+U-Boot's SPL acts a loader, loading both BL31 and BL33 (typically U-Boot).
+Loading is done from SD card, eMMC or SPI flash, also via an USB debug
+interface (FEL).
+BL31 lives in SRAM A2, which is documented to be accessible from secure
+world only.
+
+Current limitations:
+
+-  Missing PMIC support
+
+After building bl31.bin, the binary must be fed to the U-Boot build system
+to include it in the FIT image that the SPL loader will process.
+bl31.bin can be either copied (or sym-linked) into U-Boot's root directory,
+or the environment variable BL31 must contain the binary's path.
+
+To build:
+
+::
+
+    make CROSS_COMPILE=aarch64-none-elf- PLAT=sun50i_a64 DEBUG=1 bl31
```

Comment:
Actually, I hadn't seen this, is this really the toolchain that has to be used? The user guide says we support ``aarch64-linux-gnu-``, not ``aarch64-none-elf-``. I don't know what the differences are (I expect them to be minimal) but maybe it's worth changing it?

Reply:
Please don't shoot the messenger, I just copied this from QEMU, ZynqMP and Tegra ;-)
Sure can change this, I would indeed hope that people use the distribution provided cross compilers these days, which typically have a full Linux prefix.

Reply:
Oh, no, I can imagine that was the case! I created the Raspberry Pi port from Ard's work, and I'm sure that there are still some details that I haven't noticed even though I spent quite some time reviewing it to detect problems and reworking it. It's just that I was trying to compile it and it struck me as something weird. :P